configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java

```diff
@@ -719,20 +721,26 @@ private ServiceInfo getTesterServiceInfo(ApplicationId applicationId) {
 
     // ---------------- Session operations ----------------------------------------------------------------
 
-    public CompletionWaiter activate(Session session, Session previousActiveSession, ApplicationId applicationId, boolean force) {
-        CompletionWaiter waiter = session.getSessionZooKeeperClient().createActiveWaiter();
+    public Activation activate(Session session, ApplicationId applicationId, Tenant tenant, boolean force) {
         NestedTransaction transaction = new NestedTransaction();
-        transaction.add(deactivateCurrentActivateNew(previousActiveSession, session, force));
-        if (hostProvisioner.isPresent()) {
-            try (var applicationTransaction = new ApplicationTransaction(hostProvisioner.get().lock(applicationId), transaction)) {
+        Optional<ApplicationTransaction> applicationTransaction = hostProvisioner.map(provisioner -> provisioner.lock(applicationId))
```

Comment:
Consider making an AutoCloseable class to replace these lines and the tenant.getApplicationRepo().lock().

Reply:
Tried it, didn't make the flow clearer imo.